### PR TITLE
Fixed: cluster asking for a network equal to a pool runs into an error

### PR DESCRIPTION
**Pull request by davidefalcone1**
# Description

Currently, if a remote cluster asks for a network which is equal to a network pool, it gets an error. This happens because AcquireSpecificChildPrefix function returns an error if the parent and the child cidr are equal.
The same happens when reserving a network.

This PR adds an additional check in functions AcquireReservedSubnet and GetSubnetPerCluster to check this condition.

### CODE REVIEW

**Review by alacuku — APPROVED** [review]

**Review by palexster — APPROVED** [review]

File: pkg/liqonet/ipam_test.go [hunk 1]
```diff
@@ -87,42 +109,95 @@ var _ = Describe("IpamNew", func() {
...
+				It("should not allocate the network", func() {
+					_, err := ipam.GetSubnetPerCluster("10.0.0.0/9", "cluster1")
+					gomega.Expect(err).To(gomega.BeNil())
+					gomega.Expect(ipam.SubnetPerCluster["cluster1"]).To(gomega.Equal("10.0.0.0/9"))
+					_, err = ipam.GetSubnetPerCluster("10.128.0.0/9", "cluster2")
+					gomega.Expect(err).To(gomega.BeNil())
+					gomega.Expect(ipam.SubnetPerCluster["cluster2"]).To(gomega.Equal("10.128.0.0/9"))
+					_, err = ipam.GetSubnetPerCluster("10.1.0.0/16", "cluster3")
+					gomega.Expect(err).ToNot(gomega.BeNil())
+					/*err = ipam.FreeSubnetPerCluster("cluster1")
```
Comment (palexster) [inline pkg/liqonet/ipam_test.go:160]: Commented code is dead code and should be dropped or decommented

File: pkg/liqonet/ipam_test.go [hunk 2]
```diff
@@ -87,42 +109,95 @@ var _ = Describe("IpamNew", func() {
...
+					err := ipam.Init(reserved, pool, subnetPerCluster)
+					gomega.Expect(err).To(gomega.BeNil())
+				})
 				It("Should allocate the subnet itself, without mapping", func() {
 					_, err := ipam.GetSubnetPerCluster("10.0.0.0/16", "cluster1")
 					gomega.Expect(err).To(gomega.BeNil())
 					gomega.Expect(ipam.SubnetPerCluster["cluster1"]).To(gomega.Equal("10.0.0.0/16"))
-					err = ipam.FreeSubnetPerCluster("cluster1")
-					gomega.Expect(err).To(gomega.BeNil())
+					/*err = ipam.FreeSubnetPerCluster("cluster1")
```
Comment (palexster) [inline pkg/liqonet/ipam_test.go:183]: ditto

File: pkg/liqonet/ipam_test.go [hunk 3]
```diff
@@ -131,30 +206,119 @@ var _ = Describe("IpamNew", func() {
...
+						gomega.Expect(err).To(gomega.BeNil())
+					})
 					It("should allocate it as a new prefix", func() {
 						_, err := ipam.GetSubnetPerCluster("10.0.0.0/16", "cluster1")
 						gomega.Expect(err).To(gomega.BeNil())
 						gomega.Expect(ipam.SubnetPerCluster["cluster1"]).To(gomega.Equal("10.0.0.0/16"))
 						_, err = ipam.GetSubnetPerCluster("10.0.0.0/16", "cluster2")
 						gomega.Expect(err).To(gomega.BeNil())
-						err = ipam.FreeSubnetPerCluster("cluster1")
+						/*err = ipam.FreeSubnetPerCluster("cluster1")
```
Comment (palexster) [inline pkg/liqonet/ipam_test.go:234]: ditto

File: pkg/liqonet/ipam_test.go [hunk 4]
```diff
@@ -131,30 +206,119 @@ var _ = Describe("IpamNew", func() {
 						gomega.Expect(err).To(gomega.BeNil())
 						gomega.Expect(ipam.SubnetPerCluster["cluster2"]).To(gomega.HavePrefix("10."))
 						gomega.Expect(ipam.SubnetPerCluster["cluster2"]).To(gomega.HaveSuffix("/16"))
-						err = ipam.FreeSubnetPerCluster("cluster1")
+						/*err = ipam.FreeSubnetPerCluster("cluster1")
```
Comment (palexster) [inline pkg/liqonet/ipam_test.go:209]: ditto

File: pkg/liqonet/ipam_test.go (hunk 1 above)
Comment (davidefalcone1) [reply pkg/liqonet/ipam_test.go:160]: Sorry, I missed it!

**Pull request merged**